code/modules/projectiles/guns/ballistic/revolver.dm

```diff
@@ -92,48 +92,43 @@
 						"Black Panther" = "detective_panther"
 						)
 
-/obj/item/gun/ballistic/revolver/detective/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0)
-	if(magazine && magazine.caliber != initial(magazine.caliber))
+/obj/item/gun/ballistic/revolver/detective/process_fire(atom/target, mob/living/user, message = TRUE, params = null, zone_override = "", bonus_spread = 0, skip_missfire_check = FALSE)
```

Comment:
The arguments of redefined procs should be the same in parent and child.
Turn this into a detective revolver variable, that will be more maintainable.

Reply:
This is done now I believe.